File: PDOO-Civitas/CivitasJ/src/civitas/SorpresaEspeculador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 */
public class SorpresaEspeculador extends Sorpresa{
    
    private String texto;
    private int valor;
    MazoSorpresas mazo;
    
    SorpresaEspeculador(int valor, String texto){
    
        this.texto = texto;
        this.informacion = this.texto;
        this.valor = valor;
        this.mazo = new MazoSorpresas();
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
    
        if(this.jugadorCorrecto(actual, todos)){
        
            this.informe(actual, todos);
            
            JugadorEspeculador especulador = new JugadorEspeculador(todos.get(actual), this.valor);
            
            todos.remove(actual);
            todos.add(especulador);
            
        }
        
    }
    
    @Override
    void informe(int actual, ArrayList<Jugador> todos){
    
        Diario.getInstance().ocurreEvento("Se esta aplicando una sorpresa CONVERSIÓN ESPECULADOR al jugador " + todos.get(actual).getNombre());
    }
    
    @Override
    public String toString(){
        
        String sorpresa = "Tipo: Sorpresa Conversión Especulador\n";
        
        sorpresa += this.texto;
        
        return sorpresa;
    }
    
}

File: PDOO-Civitas/CivitasR/lib/sorpresa_especulador.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "sorpresa"
require_relative "jugador_especulador"

module Civitas
  class Sorpresa_especulador < Sorpresa
    
    public
    def initialize(valor, texto)
      
      @valor = valor
      @mazo = Mazo_Sorpresas.new
      @texto = texto
      
    end
    
    public
    def aplicar_a_jugador(actual, todos)
      
      if(jugador_correcto(actual, todos))
        
            informe(actual, todos)
            
            especulador = Jugador_especulador.new(todos.at(actual), @valor)
            
            todos.delete_at(actual)
            todos.push(especulador)
            
      end
      
    end
    
    private
    def informe(actual, todos)
      
      Diario.instance.ocurre_evento("Se esta aplicando una sorpresa CONVERSIÓN ESPECULADOR al jugador " + todos.at(actual).to_s);
      
    end
    
    public
    def to_s()
      
      return "Tipo: Sorpresa Conversión Especulador\n" + @texto

    end
    
    public_class_method :new

  end
end
